apps/antalmanac/src/lib/TutorialHelpers.tsx

```diff
@@ -0,0 +1,239 @@
+        window.matchMedia('(max-width: 768px)').matches ||
+        localStorage.getItem('userId') != null
+    );
+}
+
+function KbdCard(props: { children?: React.ReactNode }) {
+    return (
+        <div
+            style={{
+                borderRadius: '0.25rem',
+                border: '1px solid #ccc',
+                backgroundColor: '#f7f7f7',
+                display: 'inline',
+                verticalAlign: 'middle',
+                textAlign: 'center',
+                fontFamily: 'monospace',
+                padding: '0.05rem 0.3rem',
+            }}
+        >
+            {props.children}
+        </div>
+    );
+}
+
+export function namedStepsFactory(goToStep: (step: number) => void): Record<TourStepName, StepType> {
+    const setTab = useTabStore.getState().setActiveTab;
+
+    const goToNamedStep = (stepName: TourStepName) => {
+        const stepIndex = tourStepNames.findIndex((step) => step == stepName);
+
+        if (stepIndex == -1) {
+            console.error(`Could not find step with name ${stepName}`);
+            return;
+        }
+
+        goToStep(stepIndex);
+    };
+
+    return {
+        welcome: {
+            selector: '#root',
+            content: (
+                <>
+                    Welcome to AntAlmanac!
+                    <br />
+                    Use <KbdCard>←</KbdCard> and <KbdCard>→</KbdCard> to navigate the tour.
+                    <br />
+                    Press <KbdCard>Esc</KbdCard> to exit.
+                    <hr />
+                    You can always review the tour by clicking the button in the bottom right corner.
+                </>
+            ),
+        },
+        searchBar: {
+            selector: '#searchBar',
+            content: 'You can search for your classes here!',
+            action: () => {
+                markTourHasRun();
+                setTab(0);
+            },
+            mutationObservables: ['#searchBar'],
+        },
+        importButton: {
+            selector: '#import-button',
+            content: 'Quickly add your classes from WebReg or Zotcourse!',
+        },
+        calendar: {
+            selector: '.rbc-time-view', // Calendar.
+            content: 'See the classes in your schedule!',
+            action: addSampleClasses,
+        },
+        finalsButton: {
+            selector: '#finals-button, #finals-button-pressed',
+            content: (
+                <>
+                    <b>Click</b> to see your finals
+                </>
+            ),
+            /** If the user presses/pressed the finals button, move to the next step */
+            action: tourActionFactory(
+                () => {
+                    goToNamedStep(TourStepName.showFinals);
+                },
+                // Only move to the next step if the button is toggled from off to on, not vice versa.
+                { selector: '#finals-button', eventType: 'click' }
+            ),
+        },
+        showFinals: {
```

Comment:
Can we unclick the button after this step? Currently the tutorial won't make sense if the button is already clicked. 

In the same vein, we should set the finals button to unclicked before entering the `calendar` step, so that the appropriate calendar is shown.